discovery/module.go

```diff
@@ -115,47 +120,67 @@ func (m *Module) Add(serviceID string, presentation vc.VerifiablePresentation) e
 	if presentation.Format() != vc.JWTPresentationProofFormat {
 		return errors.New("only JWT presentations are supported")
 	}
-	// TODO: validate signature
 	if presentation.ID == nil {
 		return errors.New("presentation does not have an ID")
 	}
 	expiration := presentation.JWT().Expiration()
-	// VPs should not be valid for too short; unnecessary overhead at the server and clients.
-	// Also protects against expiration not being set at all. The factor is somewhat arbitrary.
-	minValidity := float64(definition.PresentationMaxValidity / 4)
-	if expiration.Sub(time.Now()).Seconds() < minValidity {
-		return fmt.Errorf("presentation is not valid for long enough (min %s)", time.Duration(minValidity)*time.Second)
+	if expiration.IsZero() {
+		return errors.New("presentation does not have an expiration")
 	}
 	// VPs should not be valid for too long, as that would prevent the server from pruning them.
 	if int(expiration.Sub(time.Now()).Seconds()) > definition.PresentationMaxValidity {
 		return fmt.Errorf("presentation is valid for too long (max %s)", time.Duration(definition.PresentationMaxValidity)*time.Second)
 	}
-
+	// Check if the presentation already exists
+	credentialSubjectID, err := credential.PresentationSigner(presentation)
+	if err != nil {
+		return err
+	}
+	exists, err := m.store.exists(serviceID, credentialSubjectID.String(), presentation.ID.String())
```

Comment:
mixed use of fields. above we use the `exp` from the JWT, here the ID field in the json object. why not the `jti`? Probably doesn't matter, but being consistent might save some issues in the future?

Reply:
The VP spec does not specify expiration date, so there's no property for it. Only in the JSON-LD and JWT proofs.

I think properties from the VC model should be preferred (so presentation.ID over JWT jti, although they map to the same underlying JWT field), unless there of course isn't one. Therefore it's mixed.